pihole/tests/test_pihole.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from datadog_checks.base import ConfigurationError
+from datadog_checks.pihole import PiholeCheck
+
+
+@pytest.mark.unit
+def test_config():
+    instance = {}
+    c = PiholeCheck('pihole', {}, [instance])
+
+    # empty should fail
+    with pytest.raises(ConfigurationError):
+        c.check(instance)
+
+
+# Testing integration status check using docker
+@pytest.mark.integrations
+@pytest.mark.usefixtures('dd_environment_pass')
+def test_service_check(aggregator, instance_pass):
+    c = PiholeCheck('pihole', {}, [instance_pass])
+
+    c.check(instance_pass)
+    aggregator.assert_service_check('pihole.running', PiholeCheck.OK)
+
+
+# Testing known metric value using docker
+@pytest.mark.integrations
+@pytest.mark.usefixtures('dd_environment_pass')
+def test_positive_response(aggregator, instance_pass):
+    c = PiholeCheck('pihole', {}, [instance_pass])
+
+    c.check(instance_pass)
+    aggregator.assert_metric(name="pihole.domains_being_blocked", value=125230.0)
```

Comment:
Move the metrics to an array and do
```
for metric, value in METRICS.iteritems():
    aggregator.assert_metric(name=metric, value=value)
```

Also include metric type (gauge, rate etc) on the assertion